File: Linear Data Structures/assignment/LinkedList.py
class Node:
    def __init__(self, val=None, next=None):
        self.val = val
        self.next = next

class LinkedList:
    def __init__(self, head=None):
        self.head = head

    # Get the element at certain index, and
    # return the value of the element
    def getElement(self, idx):
        pass

    # Search if the element exists in the LinkedList
    # return True if element exist, else False
    def searchElement(self, n):
        pass
    
    # Insert the new element to the front of the LinkedList
    # to achieve O(1) time complexity
    # Can we do an insert back with O(1) time complexity? If so, how?
    # You are welcome to implement it. You can make changes to the constructor as well
    def insertFront(self, n):
        pass

    # Remove the element from the LinkedList
    def remove(self, n):
        pass

    # Print all of the elements in the LinkedList
    def __str__(self):
        return

File: Linear Data Structures/assignments/Java/LinkedList.java
public class LinkedList {
    
    static class Node {
        Object value;
        Node next;

        Node (Object value) {
            this.value = value;
        }
    }

    private Node head;

    // Constructor
    public LinkedList() {
        this.head = null;
    }


    /*
     * Get the value of the element at certain index
     * @return value of the index
     */
    public Object getElement(int idx) {
        // Code here
        return this.head.value; // Just a temporary placeholder
    }

    
    /*
     * Search if the element exist in the LinkedList
     * @return True if the element exist, else False
     */
    public boolean searchElement(Object n) {
        // Code here
        return true; // Just a temporary placeholder
    }

    /*
     * Insert the value at the front of the LinkedList
     * Can we do an insert back with O(1) time complexity? If so, how?
     * You are welcome to implement it
     */
    public void insertFront(Object n) {
        // Code here
    }


    /*
     * Remove the element from LinkedList
     */
    public void removeElement(Object n) {
        // Code here
    }


    /*
     * Additional function (optional implementation). This is an interview question commonly asked
     * by big tech companies (Google, Facebook, Bloomberg, Amazon). e.g:
     * Your current Linked List: 5 -> 4 -> 3 -> 2 -> 1 -> None
     * After calling the function: 1 -> 2 -> 3 -> 4 -> 5 -> None
     */
    public void reverseLinkedList() {
        // Code here
    }

    @Override
    /*
     * Printing out all of the value in the LinkedList
     */
    public String toString() {
        // Code here
        return ""; // Just a temporary placeholder
    }
}